packages/devextreme/js/__internal/ui/splitter/splitter.ts

```diff
@@ -804,10 +804,18 @@ class Splitter extends CollectionWidget<Properties> {
         : this._calculateExpandToRightSize(itemIndex + 1);
     }
 
-    const itemSize = targetPaneSize >= minSize
+    let itemSize = targetPaneSize >= minSize
       ? targetPaneSize
       : minSize;
 
+    if (isDefined(item.maxSize)) {
+      const { orientation } = this.option();
+      const handlesSizeSum = this._getResizeHandlesSize();
+      const elementSize = getElementSize($(this.element()), orientation);
```

Comment:
I believe we can avoid getting the element sizes twice during collapsing. I will try to refactor this part in the next commit